otx/algorithms/action/adapters/mmaction/data/dataset.py

```diff
@@ -0,0 +1,146 @@
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+
+from copy import copy
+from typing import Any, Dict, List, Sequence
+
+import numpy as np
+from mmaction.datasets.builder import DATASETS
+from mmaction.datasets.pipelines import Compose
+from mmaction.datasets.rawframe_dataset import RawframeDataset
+
+from otx.api.entities.datasets import DatasetEntity
+from otx.api.entities.label import LabelEntity
+from otx.api.utils.argument_checks import (
+    DatasetParamTypeCheck,
+    check_input_parameters_type,
+)
+
+
+# pylint: disable=too-many-instance-attributes
+@DATASETS.register_module()
+class OTXRawframeDataset(RawframeDataset):
+    """Wrapper that allows using a OTX dataset to train mmaction models.
+
+    This wrapper is not based on the filesystem,
+    but instead loads the items here directly from the OTX DatasetEntity object.
+    """
+
+    class _DataInfoProxy:
+        def __init__(self, otx_dataset, labels):
+            self.otx_dataset = otx_dataset
+            self.labels = labels
+            self.label_idx = {label.id: i for i, label in enumerate(labels)}
+
+        def __len__(self):
+            return len(self.otx_dataset)
+
+        def __getitem__(self, index):
+            """Prepare a dict 'data_info' that is expected by the mmaction pipeline to handle images and annotations.
+
+            :return data_info: dictionary that contains the image and image metadata, as well as the labels of
+            the objects in the image
+            """
+
+            dataset = self.otx_dataset
+            item = dataset[index]
+            ignored_labels = np.array([self.label_idx[lbs.id] for lbs in item.ignored_labels])
+
+            data_info = dict(
+                dataset_item=item,
+                index=index,
+                ann_info=dict(label_list=self.labels),
+                ignored_labels=ignored_labels,
+            )
+
+            return data_info
+
+    @check_input_parameters_type({"otx_dataset": DatasetParamTypeCheck})
+    # pylint: disable=too-many-arguments, invalid-name, super-init-not-called
+    # TODO Check need for additional params such as multi_class, with_offset
+    def __init__(
+        self,
+        otx_dataset: DatasetEntity,
+        labels: List[LabelEntity],
+        pipeline: Sequence[dict],
+        data_prefix=str,
+        test_mode: bool = False,
+        filename_tmpl: str = "img_{:05}.jpg",
+        start_index: int = 1,
+        modality: str = "RGB",
+    ):
+        self.otx_dataset = otx_dataset
+        self.labels = labels
+        self.data_prefix = data_prefix
+        self.test_mode = test_mode
```

Comment:
Seems like `self.test_mode` is not used anywhere.

Reply:
Maybe we can use `self.test_mode` on [line#141](https://github.com/openvinotoolkit/training_extensions/pull/1328/files#diff-bf927e97fa4cb305e8cc97a2eab9de9b54ce7fdcd059f3efb89496974df5d9ceR141)

Reply:
It is used inside of mmaction2